File: ResLogger2.Web/services/web/project/exists_result.py


class ExistsResult:
    def __init__(self, index_id: int = -1, full_text: str = "", full_hash: int = -1, full_exists: bool = False):
        self.index_id = index_id
        self.full_text = full_text
        self.full_hash = full_hash
        self.full_exists = full_exists

    def file_text(self) -> str:
        return self.full_text[self.full_text.rfind("/"):]

    def folder_text(self) -> str:
        return self.full_text[:self.full_text.rfind("/")+1]

File: ResLogger2.Plugin/ExistsResult.cs
namespace ResLogger2.Plugin;

public struct ExistsResult
{
    public uint IndexId;
    
    public string FileText => FullText[(FullText.LastIndexOf('/') + 1)..];
    public string FolderText => FullText[..FullText.LastIndexOf('/')];
    public string FullText;
            
    public uint FileHash => Lumina.Misc.Crc32.Get(FileText);
    public uint FolderHash => Lumina.Misc.Crc32.Get(FolderText);
    public uint FullHash => Lumina.Misc.Crc32.Get(FullText);
    
    public bool FullExists;

    public override bool Equals(object obj)
    {
        return obj is ExistsResult er &&
               IndexId == er.IndexId &&
               FolderText.Equals(er.FolderText);
    }

    public override int GetHashCode()
    {
        return unchecked((int) FullHash);
    }

    public override string ToString()
    {           
        return $"Path: {FullText} Exists: {FullExists} In: {IndexId:X6}";                
    }
}